karapace/schema_reader.py

```diff
@@ -487,7 +487,7 @@ def _handle_msg_config(self, key: dict, value: dict | None) -> None:
     def _handle_msg_delete_subject(self, key: dict, value: dict | None) -> None:  # pylint: disable=unused-argument
         if value is None:
             LOG.warning("DELETE_SUBJECT record doesnt have a value, should have")
-            return
+            raise ValueError("DELETE_SUBJECT record doesnt have a value, should have")
```

Comment:
This will not get picked up by the error handling in `handle_messages` but will get handled as unexpected exception in `run`.
You could add a one more test case in the error handling tests.

Reply:
It's already handled, check this line - https://github.com/Aiven-Open/karapace/blob/main/karapace/schema_reader.py#L606, we catch this and raise `InvalidSchema`, there's also a test for this.

Reply:
Good, although the message in the exception is not correct.

Reply:
Yes I agree, i tried not to change most of these.